webapp/app.py

```diff
@@ -22,3 +25,32 @@
 template_finder_view = TemplateFinder.as_view("template_finder")
 app.add_url_rule("/", view_func=template_finder_view)
 app.add_url_rule("/<path:subpath>", view_func=template_finder_view)
+
+
+def static_url(filename):
+    """
+    Template function for generating URLs to static assets:
+    Given the path for a static file, output a url path
+    with a hex hash as a query string for versioning
+    """
+
+    filepath = os.path.join("static", filename)
+    url = "/" + filepath
+
+    if not os.path.isfile(filepath):
+        # Could not find static file
+        return url
+
+    # Use MD5 as we care about speed a lot
+    # and not security in this case
+    file_hash = hashlib.md5()
+    with open(filepath, "rb") as file_contents:
+        for chunk in iter(lambda: file_contents.read(4096), b""):
+            file_hash.update(chunk)
+
+    return url + "?v=" + file_hash.hexdigest()[:7]
+
+
+@app.context_processor
+def utility_processor():
```

Comment:
same for this: https://github.com/canonical-web-and-design/canonicalwebteam.flask-base/blob/8e303653e442d87937a1c89398d0594f17ecbf73/canonicalwebteam/flask_base/context.py#L39

Reply:
Oh! Great. thanks!